### PR TITLE
Dismiss search bar if user click outside that element

### PR DESCRIPTION
**Pull request by FadhlanR**
Ticket: CS-5952

### CODE REVIEW

File: packages/host/app/components/search-sheet/index.gts [hunk 1]
```diff
@@ -60,6 +62,15 @@ export default class SearchSheet extends Component<Signature> {
   @service declare cardService: CardService;
   @service declare loaderService: LoaderService;
 
+  constructor(args: any, owner: any) {
+    super(args, owner);
+
+    document.addEventListener('click', this.onMouseClick);
```
Comment (jurgenwerk) [inline packages/host/app/components/search-sheet/index.gts]: Did you consider making a [modifier](https://github.com/ember-modifier/ember-modifier)? I think adding custom event handlers with side effects could be a good fit for one of those

Comment (jurgenwerk) [reply packages/host/app/components/search-sheet/index.gts]: Actually, there is an addon that it looks like it does exactly what is implemented in this PR, with a modifier too: https://github.com/zeppelin/ember-click-outside

However I'm not sure if it's worth adding this addon and risk future tech debt for such a simple thing. Maybe we can just copy the approach.

Comment (lukemelia) [reply packages/host/app/components/search-sheet/index.gts]: ember-click-outside is well-implemented and well-maintained. +1 to using it.

Comment (FadhlanR) [reply packages/host/app/components/search-sheet/index.gts]: Updated [b4feca9](https://github.com/cardstack/boxel/pull/625/commits/b4feca9d46f5ce00ac8c55dad1d293cc89823734)

**Review by jurgenwerk — APPROVED** [review]

File: packages/host/app/components/search-sheet/index.gts [hunk 2]
```diff
@@ -226,7 +228,12 @@ export default class SearchSheet extends Component<Signature> {
   }
 
   <template>
-    <div class='search-sheet {{this.sheetSize}}' data-test-search-sheet>
+    <div
+      id='search-sheet'
+      class='search-sheet {{this.sheetSize}}'
+      data-test-search-sheet={{@mode}}
+      {{onClickOutside @onBlur exceptSelector='.add-card-to-neighbor-stack'}}
```
Comment (habdelra) [inline packages/host/app/components/search-sheet/index.gts:235]: cool!

**Review by habdelra — APPROVED** [review]

**Pull request merged**